static/app/components/charts/baseChart.tsx

```diff
@@ -435,90 +448,120 @@ function BaseChartUnwrapped({
       ? getDiffInMinutes({start, end, period}) <= minutesThresholdToDisplaySeconds
       : false;
 
-  const xAxisOrCustom = !xAxes
-    ? xAxis !== null
-      ? XAxis({
-          ...xAxis,
-          theme,
-          useShortDate,
-          start,
-          end,
-          period,
-          isGroupedByDate,
-          addSecondsToTimeFormat,
-          utc,
-        })
-      : undefined
-    : Array.isArray(xAxes)
-    ? xAxes.map(axis =>
-        XAxis({
-          ...axis,
-          theme,
-          useShortDate,
-          start,
-          end,
-          period,
-          isGroupedByDate,
-          addSecondsToTimeFormat,
-          utc,
-        })
-      )
-    : [XAxis(defaultAxesProps), XAxis(defaultAxesProps)];
-
-  const seriesData =
-    Array.isArray(series?.[0]?.data) && series[0].data.length > 1
-      ? series[0].data
-      : undefined;
-  const bucketSize = seriesData ? seriesData[1][0] - seriesData[0][0] : undefined;
-
   const isTooltipPortalled = tooltip?.appendToBody;
 
-  const tooltipOrNone =
-    tooltip !== null
-      ? ChartTooltip({
-          showTimeInTooltip,
-          isGroupedByDate,
-          addSecondsToTimeFormat,
-          utc,
-          bucketSize,
-          ...tooltip,
-          className: isTooltipPortalled
-            ? `${tooltip?.className ?? ''} chart-tooltip-portal`
-            : tooltip?.className,
-        })
-      : undefined;
-
-  const aria = useEchartsAriaLabels(
-    {
+  const chartOption = useMemo(() => {
+    const seriesData =
+      Array.isArray(series?.[0]?.data) && series[0].data.length > 1
+        ? series[0].data
+        : undefined;
+
+    const bucketSize = seriesData ? seriesData[1][0] - seriesData[0][0] : undefined;
+    const tooltipOrNone =
+      tooltip !== null
```

Comment:
Not equal covers undefined as well:

```suggestion
      tooltip != null
```

Reply:
It seems like we allow undefined as a value, so I'm not sure if this was intentional as I just copied it over